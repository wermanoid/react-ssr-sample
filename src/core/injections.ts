import { inject } from 'inversify';

export const property = <A extends {} | Function>(abstraction: A) => <
  T,
  C extends keyof T & string
>(
  target: T,
  property: C,
  index?: number
) => {
  const injectionName =
    (<Function>abstraction).name || abstraction.constructor.name;
  inject(injectionName)(target, property, index);
};

export const argument = property;

export const method = <A extends {}>(...abstractions: A[]) => (
  target: {},
  name: string,
  descriptor: {}
) => {
  console.log(target, name, descriptor);
};
