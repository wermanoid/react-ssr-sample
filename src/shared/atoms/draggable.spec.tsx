import { mount } from 'enzyme';
import React from 'react';
import styled from '@emotion/styled';
import { css } from '@emotion/core';

import Draggable from './draggable';

const Btn = styled.button<{ width: number }>`
  color: red;
  ${props => css`
    width: ${props.width};
  `};
`;

const Sample = () => (
  <div css={{ background: 'rgba(255,255,255, 0.4)' }}>
    <Btn width={234}>lol</Btn>
  </div>
);

describe('sample', () => {
  it('sjoudl', () => {
    const wrap = mount(
      <Draggable id="324" render="div">
        <Sample />
      </Draggable>
    );
    expect(wrap).toMatchSnapshot();
  });
});
