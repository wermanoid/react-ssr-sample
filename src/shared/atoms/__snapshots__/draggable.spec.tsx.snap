// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sample sjoudl 1`] = `
.emotion-2 {
  background: rgba(255,255,255,0.4);
}

.emotion-0 {
  color: red;
  width: 234;
}

<Draggable
  id="324"
  render="div"
>
  <div
    draggable={true}
    onDragStart={[Function]}
  >
    <Sample>
      <div
        className="emotion-2"
      >
        <Btn
          width={234}
        >
          <button
            className="emotion-0 emotion-1"
            width={234}
          >
            lol
          </button>
        </Btn>
      </div>
    </Sample>
  </div>
</Draggable>
`;
